<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<h3>Analyst: Kyla Caliwan</h3>

In [ ]:
#import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
from scipy import stats

<h2>Converting Dataset(s) into Dataframe</h2>

In [ ]:
penguins = pd.read_csv('datasets//penguins.csv')

In [27]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [28]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [29]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [30]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [31]:
penguins['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

In [32]:
penguins['mass_kg'] = np.round(penguins['body_mass_g']/1000,1)

In [33]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[344 rows x 9 columns]

<h1>Exploratory Data Analysis (EDA)</h1>
<hr>
<h2>Question No.1</h2>
<h3> What is the most common sex among Gentoo penguins?</h3>

In [34]:
penguins[penguins['species'] == 'Gentoo']['sex']

152    female
153      male
154    female
155      male
156      male
        ...  
271       NaN
272    female
273      male
274    female
275      male
Name: sex, Length: 124, dtype: object

In [35]:
penguins[penguins['species'] == 'Gentoo']['sex'].mode()[0]

'male'

<h3>Insight: The most common sex among Gentoo penguins is male.</h3>

<h2>Question No.2</h2>
<h3> Which island has the largest average flipper length?</h3>

In [36]:
penguins.groupby('island')['flipper_length_mm'].mean()

island
Biscoe       209.706587
Dream        193.072581
Torgersen    191.196078
Name: flipper_length_mm, dtype: float64

<h3>Insight: The island with the largest average flipper length is Biscoe, who got 209.706587.</h3>

<h2>Question No.3</h2>
<h3> How many penguins are not from 'Biscoe' island?</h3>

In [37]:
(penguins['island'] != 'Biscoe')

0      True
1      True
2      True
3      True
4      True
       ... 
339    True
340    True
341    True
342    True
343    True
Name: island, Length: 344, dtype: bool

In [38]:
(penguins['island'] != 'Biscoe').sum()

np.int64(176)

<h3>Insight: There are 176 penguins not from Biscoe island.</h3>

<h2>Question No.4</h2>
<h3>Which species has the lowest body mass? </h3>

In [39]:
penguins['body_mass_g'].min()

np.float64(2700.0)

In [40]:
penguins[penguins['body_mass_g']== 2700.0]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
314  Chinstrap  Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  mass_kg  
314       2700.0  female  2008      2.7

<h3>Insight : The species with the lowest average body mass is Chinstrap.</h3>

<h2>Question No.5</h2>
<h3>What is the maximum bill length for penguins from the 'Dream' island?</h3>

In [41]:
penguins[penguins['island'] == 'Dream']['bill_length_mm'].max()

np.float64(58.0)

In [42]:
penguins[penguins['bill_length_mm']== 58.0]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
293  Chinstrap  Dream            58.0           17.8              181.0   

     body_mass_g     sex  year  mass_kg  
293       3700.0  female  2007      3.7

<h3>Insight: The maximum bill length for penguins from Dream island is 58.0 mm, recorded from a Chinstrap penguin.</h3>

<h2>Question No.6</h2>
<h3>Which species has the most males?</h3>

In [43]:
penguins[penguins['sex'] == 'male']['species'].value_counts()

species
Adelie       73
Gentoo       61
Chinstrap    34
Name: count, dtype: int64

In [44]:
penguins[penguins['sex'] == 'male']['species'].mode()[0]

'Adelie'

<h3>Insight: The species with the most males is Adelie, with 73 recorded</h3>

<h2>Question No.7</h2>
<h3>Which sex has the highest average bill length among Adelie penguins?</h3>

In [45]:
penguins[penguins['species'] == 'Adelie'].groupby('sex')['bill_length_mm'].mean()

sex
female    37.257534
male      40.390411
Name: bill_length_mm, dtype: float64

<h3>Insight: Among Adelie penguins, males have the highest average bill length.</h3>

<h2>Question No.8</h2>
<h3>What percentage of penguins in the dataset are male vs female?</h3>

In [50]:
(penguins["sex"].value_counts(normalize=True) * 100).round(2)

sex
male      50.45
female    49.55
Name: proportion, dtype: float64

<h3>Insight: The highest percentage distribution of penguins in dataset is male penguins that got 50.45%</h3>

<h2>Question No.9</h2>
<h3>How many penguins there are per species.</h3>

In [55]:
penguins.groupby("species").size()


species
Adelie       152
Chinstrap     68
Gentoo       124
dtype: int64

<h3>Insight: The highest most populated species of penguins are Adelie with a count of 152</h3>

<h2>Question No.10</h2>
<h3>Which species has the smallest average bill depth?</h3>

In [28]:
penguins.groupby("species")["bill_depth_mm"].mean().idxmin()


'Gentoo'

<h3>Insights: The species that has smallest bill depth was "Gento"</h3>

<h2>Question No.11</h2>
<h3>For each species, is there a correlation between bill length and flipper length?</h3>

In [31]:
penguins.groupby("species").apply(lambda g: g["bill_length_mm"].corr(g["flipper_length_mm"]))


species
Adelie       0.325785
Chinstrap    0.471607
Gentoo       0.661162
dtype: float64

<h3>Insights: Yes, there is a correlation for all species, especially for Gentoo penguins but its weakest for Adelie penguins.</h3>

<h2>Question No.12</h2>
<h3>For each island, is there a correlation between flipper length and body mass?</h3>

In [33]:
penguins.groupby("island").apply(lambda g: g["flipper_length_mm"].corr(g["body_mass_g"]))


island
Biscoe       0.876943
Dream        0.525064
Torgersen    0.435911
dtype: float64

<h3>Insights: Yes, there is a correlation on all islands, but it’s extremely strong for Biscoe and weaker for Dream and Torgersen.</h3>

<h2>Question No.13</h2>
<h3>For each sex, is there correlation between bill depth and body mass?</h3>

In [35]:
penguins.groupby("sex").apply(lambda g: g["bill_depth_mm"].corr(g["body_mass_g"]))


sex
female   -0.747770
male     -0.755492
dtype: float64

<h3>Insights: Yes, there is a strong negative correlation for both sexes. Penguins who has the deeper bills tend to be lighter, and penguins with shallower bills tend to be heavier regardless of their sex.</h3>

In [34]:
penguins.groupby("sex").apply(lambda g: g["bill_depth_mm"].corr(g["body_mass_g"]))


sex
female   -0.747770
male     -0.755492
dtype: float64